lib/IRGen/GenMeta.cpp

```diff
@@ -4431,7 +4431,11 @@ llvm::Value *irgen::emitHeapMetadataRefForHeapObject(IRGenFunction &IGF,
     return emitLoadOfHeapMetadataRef(IGF, object,
                                      getIsaEncodingForType(IGF.IGM, objectType),
                                      suppressCast);
-
+  // Is it an opened class existential?
+  if (objectType.isAnyClassReferenceType() && isa<ArchetypeType>(objectType))
```

Comment:
I actually completely forgot about tagged pointers.  Yes, you will need to check for them dynamically.

A more efficient and idiomatic way to do this check would be to dyn_cast to ArchetypeType and call requiresClass().

Note that the code you've written isn't specific to opened class existentials — it'll actually trigger on *any* class-bounded archetype.  That's probably okay, but the comment is wrong.